python/graphstorm/run/gsgnn_np/gsgnn_np.py

```diff
@@ -87,10 +89,19 @@ def main(config_args):
         tracker.log_params(config.__dict__)
     trainer.setup_task_tracker(tracker)
     device = 'cuda:%d' % trainer.dev_id
-    dataloader = GSgnnNodeDataLoader(train_data, train_data.train_idxs, fanout=config.fanout,
-                                     batch_size=config.batch_size, device=device, train_task=True)
     val_dataloader = None
     test_dataloader = None
+    if config.semi_supervised:
+        # treat val and test sets as unlabeled node sets
+        unlabeled_idxs = combine_idxs(train_data.val_idxs, train_data.test_idxs)
```

Comment:
why not use the other unlabeled nodes? in many datasets, there are many unlabeled nodes that are not in the training, validation and test datasets.

Reply:
I don't think there is any unlabeled nodes in public graphs (arxiv, products, MovieLens). All target nodes . Do you think we should use all nodes not in `train_idxs` instead?

Reply:
yes, there are many datasets with nodes that are not included in training/val/test sets.